prow/config.yaml

```diff
@@ -635,3 +635,7 @@ presets:
       hostPath:
         path: /sys/fs/cgroup
         type: Directory
+# enable GOPROXY by default
```

Comment:
I don't think we support a 'default' preset that applies to all jobs implicitly and I'm not sure that we'd want to since that would be pretty opaque to job authors.
https://github.com/kubernetes/test-infra/blob/c042e2098c17d798b0078aadc3966c499c52d4dc/prow/config/jobs.go#L42-L45

Reply:
FWIW:
- presets upstream are meant to be opaque for injecting environment (primarily credential) details
- this setting only affects jobs that use go modules directly from within the container (not bazel or docker in docker)
- in many settings this would be configured with a default on the machine (EG a corporate GOPROXY)

Setting this by default across the board makes sense, I did mention in previous discussion that the prow owners may not be OK with that though 🤷‍♂

Reply:
I can confirm that the code as it is today will apply this "default" preset to every job (CI actually errors if this is set in the job spec because it merges and detects the duplicate).

+1 to everything Ben said. I can't think of any job on prow.k8s.io that wouldn't either benefit from this being set, or disregard it as a noop.

Reply:
I am fine with this, but let's make sure it's documented:
- https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#presets - mention that defaulting is possible
- https://github.com/kubernetes/test-infra/tree/master/config/jobs#job-presets - link to the default we use and mention it sets GOPROXY by default
- I'm assuming we have tests that exercise this defaulting behavior?

Reply:
Created https://github.com/kubernetes/test-infra/issues/13665 for follow up